### PR TITLE
Add new profile to create self hosted runners using github app

### PR DESCRIPTION
**Pull request by jfroche**
Copy nixos module and adapt it to use short lived github app access token to register github self hosted runners

### CODE REVIEW

**Comment by zimbatm** [comment]
Is there anything missing from this PR?

**Comment by jfroche** [comment]
nop it is working, will rebase and merge tomorrow
I ll push it in nixpkgs for review as well

File: modules/github-runners.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{ config
```
Comment (zimbatm) [inline modules/github-runners.nix:1]: this file could be moved to modules/github-runnders/default.nix

File: default.nix [hunk 2]
```diff
@@ -16,6 +16,9 @@
   mixins-systemd-boot = import ./mixins/systemd-boot.nix;
   mixins-telegraf = import ./mixins/telegraf.nix;
 
+  # Modules
+  github-runner = import ./modules/github-runners.nix;
```
Comment (zimbatm) [inline default.nix:20]: If the module is getting upstreamed, let's keep it private and import it in the role directly.

**Review by zimbatm — COMMENTED** [review]
looking good!

**Review by zimbatm — APPROVED** [review]

**Pull request merged**